### PR TITLE
Added valid_move? for King and tests for valid and invalid movements

### PR DESCRIPTION
**Pull request by Brandongoodman615**
valid_move? for King is working with tests. 
Made minor adjustment for is_obstructed test.

### CODE REVIEW

File: spec/models/king_spec.rb [hunk 1]
```diff
@@ -1,5 +1,28 @@
 require 'rails_helper'
 
 RSpec.describe King, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe "is_valid_move king" do 
+    it "should be an invalid movement" do
```
Comment (ilyakrasnov) [inline spec/models/king_spec.rb]: @Brandongoodman615 awesome tests! 👍

File: app/models/king.rb [hunk 2]
```diff
@@ -1,2 +1,7 @@
 class King < Piece
+
+  def valid_move?(x,y)
+    y > y_coord + 1 || y < y_coord - 1 || x > x_coord + 1 || x < x_coord - 1 ? false : true
```
Comment (ilyakrasnov) [inline app/models/king.rb]: What do `y > y_coord + 1` and `y < y_coord - 1` and same for `x` have in common? Thinking about finding a way to combine them in one statement.. Any ideas 😄

Comment (ilyakrasnov) [reply app/models/king.rb]: Another thing: what if we put in `(-10,99)`? maybe we should also handle the case of being off board?

Comment (Brandongoodman615) [reply app/models/king.rb]: @ilyakrasnov I'm just not seeing it. I've fixed for cases of moving off the board but it just made for more || statements. I just can't figure out how to combine them when the cases are specific to each variable used.

**Comment by Brandongoodman615** [comment]
Cleaned up is_obstructed method.
Added a few helper methods to piece.rb: 
- valid_board_space: to check the move is actually on the board
- square_is_occupied: for move_to method 
- x_range and y_range to clean up repeating

**Pull request merged**